concord-server/server/src/main/java/com/rabidgremlin/concord/resources/PhrasesResource.java

```diff
@@ -183,16 +185,32 @@ public Response purgeCompletedPhrasesAndVotes(@ApiParam(hidden = true) @Auth Cal
   // TODO: Create proper model class for incoming label
   public Response voteForPhrase(@ApiParam(hidden = true) @Auth Caller caller, @PathParam("phraseId") String phraseId, Label label)
   {
-    log.info("{} casting vote for {}", caller, phraseId);
+    String labelText = label.getLabel();
+    log.info("{} casting vote for {} as {}", caller, phraseId, labelText);
 
-    votesDao.upsert(phraseId, label.getLabel(), caller.getToken());
+    votesDao.upsert(phraseId, labelText, caller.getToken());
 
     // are we marking trashed phrases as completed?
-    if (completeOnTrash && StringUtils.equals(LABEL_TRASH, label.getLabel()))
+    if (completeOnTrash && StringUtils.equals(LABEL_TRASH, labelText))
     {
-      phrasesDao.markPhrasesComplete(Collections.singletonList(phraseId), Collections.singletonList(label.getLabel()));
+      phrasesDao.markPhraseComplete(phraseId, LABEL_TRASH);
     }
 
     return Response.created(uriInfo.getAbsolutePath()).build();
   }
+
+  @POST
+  @Timed
+  @Path("/{phraseId}/resolve")
+  public Response resolvePhrase(@ApiParam(hidden = true) @Auth Caller caller, @PathParam("phraseId") String phraseId, Label label)
+  {
+    String labelText = label.getLabel();
+    log.info("{} resolving {} as {}", caller, phraseId, labelText);
+
+    votesDao.upsert(phraseId, labelText, PHRASE_RESOLVER_USER_ID);
+    phrasesDao.markPhraseComplete(phraseId, labelText);
```

Comment:
This won't work, when we download the phrases (in `downloadCsv`) we only consider phrases which are not marked complete.